pkg/controller/pravega/pravega_controller.go

```diff
@@ -120,6 +120,10 @@ func MakeControllerConfigMap(p *api.PravegaCluster) *corev1.ConfigMap {
 		"-Xms1g",
 		"-XX:+UnlockExperimentalVMOptions",
 		"-XX:+UseCGroupMemoryLimitForHeap",
+		"-XX:+ExitOnOutOfMemoryError",
+		"-XX:+CrashOnOutOfMemoryError",
+		"-XX:+HeapDumpOnOutOfMemoryError",
+		"-XX:MaxRAMFraction=1",
```

Comment:
One potential problem for `-XX:MaxRAMFraction=1`  is discussed [here](https://medium.com/adorsys/jvm-memory-settings-in-a-container-environment-64b0840e1d9e#fn10-20439).
The risk of setting this to be 1 is that when under heavy load containers can get killed by k8s because of JVM tries to use up all the RAM for heap. There are other processes in the container as well as JVM overhead, so it can cause container to be OOM.
One solution is to set it to be 2, the disadvantage is that the mem utilization is 50%.
The better solution is to use `XX:MaxRAMPercentage `, the problem is that it only supported in Java 10 and above.

Reply:
Good catch @Tristan1900. Actually, we have already seen _OOMKilled_ (ref: https://github.com/pravega/pravega-operator/issues/141) when using default resource limits. My take is that we set `XX:MaxRAMFraction` to `2` until Pravega supports Java 10+. It's not optimal, but at least it should prevent pods from being _OOMKilled_ when they are under heavy load.

Reply:
Thanks! That sounds good to me.

Reply:
Just to make sure. To prevent `java.lang.OutOfMemoryError: Java heap space` in #141, we still need to set memory limit high enough. Setting XX:MaxRAMFraction to 2 only guarantee containers don't get killed by k8s silently.